### PR TITLE
Integration tests for unit properties

### LINKED ISSUE #401: Add integration tests for unit properties
### Component(s)
- hirte
- hirte-agent

### Feature under test
Getting and setting properties for a unit on an agent

### Test cases

The following sections contain proposals for test cases. 

At the moment of writing this issue, `hirtectl` doesn't provide a command to get/set properties for specific units. Therefore, its probably easier to add [pyhirte](https://pypi.org/project/pyhirte/) as a dependency and use it. 
An example for getting and setting unit properties can found [here](https://hirte.readthedocs.io/en/latest/api/bindings/), which updates the `CPUWeight` of a service. 

| ID | Test step(s) | Expected result  |
| --- |---|---|
| 1 | Get all properties of a specific unit (using `GetUnitProperties` method) | A list of properties |
| 2 | Get a property of a specific unit (using `GetUnitProperty` method) | The value of a specific property |
| 3 | Get all properties of a unit via `GetUnitProperties`, then updating a subset of them via `SetUnitProperties`, querying them again  | values of the updated subset have been updated |

### PR DESCRIPTION
**Pull request by engelmi**
This PR fixes #401 by adding integration tests for
- getting a unit property
- getting a list of all unit properties
- and setting unit property value

### CODE REVIEW

File: tests/tests/tier0/property-get/test_property_get.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+from bluechi_test.config import BluechiControllerConfig, BluechiNodeConfig
+
+
+def exec(ctrl: BluechiControllerContainer, nodes: Dict[str, BluechiNodeContainer]):
+    result, output = ctrl.run_python(os.path.join("python", "get_property.py"))
+    if result != 0:
+        raise Exception(output)
+
+
+@pytest.mark.timeout(10)
```
Comment (dougsland) [inline tests/tests/tier0/property-get/test_property_get.py]: I would expect a larger timeout if possible, like 40 as a "new default".

Comment (dougsland) [reply tests/tests/tier0/property-get/test_property_get.py]: /cc @Yarboa

File: tests/tests/tier0/properties-get/test_properties_get.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+from bluechi_test.config import BluechiControllerConfig, BluechiNodeConfig
+
+
+def exec(ctrl: BluechiControllerContainer, nodes: Dict[str, BluechiNodeContainer]):
+    result, output = ctrl.run_python(os.path.join("python", "get_properties.py"))
+    if result != 0:
+        raise Exception(output)
+
+
+@pytest.mark.timeout(10)
```
Comment (dougsland) [inline tests/tests/tier0/properties-get/test_properties_get.py]: same here

File: tests/tests/tier0/property-set/test_property_set.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    nodes[node_name_foo].copy_systemd_service(
+        service_simple, "systemd", os.path.join("/", "etc", "systemd", "system"))
+    assert nodes[node_name_foo].wait_for_unit_state_to_be(service_simple, "inactive")
+
+    result, output = ctrl.run_python(os.path.join("python", "set_property.py"))
+    if result != 0:
+        raise Exception(output)
+
+
+@pytest.mark.timeout(10)
```
Comment (dougsland) [inline tests/tests/tier0/property-set/test_property_set.py]: same here

**Review by dougsland — APPROVED** [review]
nit comments, LGTM

**Comment by dougsland** [comment]
It's always nice to see bluechi python binding being used a bunch!  🚀

File: tests/tests/tier0/property-get/test_property_get.py (hunk 1 above)
Comment (engelmi) [reply tests/tests/tier0/property-get/test_property_get.py]: @dougsland 
As I've written here: https://github.com/containers/bluechi/pull/516#issuecomment-1702244638
I'll create a PR later and align all timeouts and introduce a scoped timeout for smaller code parts.

Comment (engelmi) [reply tests/tests/tier0/property-get/test_property_get.py]: Set it to 40 for now.

**Pull request merged**